File: webapp/backend-api/backend-api/Models/Achievement.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace backend_api.Models
{
    public class Achievement : BaseEntity
    {
        //TODO: add image
        public string Name { get; set; }
        public string Description { get; set; }
        [JsonIgnore]
        public long? EventId { get; set; }
        public Event Event { get; set; }
        [JsonIgnore]
        public long? TeamId { get; set; }
        public Team Team { get; set; }
        public string CreatedBy { get; set; }
        public string Tier { get; set; }
        public string IconUrl { get; set; }
        public List<User> AchievedBy { get; set; }

    }
}

File: webapp/achievements-app/src/app/interfaces/Achievements/Achievement.ts
import { OrganizedEvent } from "../Events/OrganizedEvent"
import { Team } from "../Teams/Team"

export class Achievement{
    id: number
    name : string
    description : string
    eventId: number
    event: OrganizedEvent
    teamId: number
    team: Team
    created: Date
    createdBy: number
    tier: string
    achievedBy: []
}
